rest/sgcollect.go

```diff
@@ -82,9 +86,10 @@ func (sg *sgCollect) Start(zipFilename string, params sgCollectOptions) error {
 	args = append(args, "--sync-gateway-executable", sgPath)
 	args = append(args, zipPath)
 
-	ctx, cancelFunc := context.WithCancel(context.Background())
-	sg.cancel = cancelFunc
-	cmd := exec.CommandContext(ctx, sgCollectPath, args...)
+	sg.context = context.WithValue(context.Background(), base.LogContextKey{}, base.LogContext{CorrelationID: fmt.Sprintf("SGCollect-%s", strconv.Itoa(rand.Intn(65536)))})
+
+	sg.context, sg.cancel = context.WithCancel(sg.context)
```

Comment:
A couple of nitpicks, but also want to test the new multiline suggestions:

```suggestion
	ctx := context.WithValue(context.Background(), base.LogContextKey{}, base.LogContext{CorrelationID: "SGCollect-" + base.GenerateRandomID()})

	sg.context, sg.cancel = context.WithCancel(ctx)
```

Reply:
Haha at the multiline suggestions...
Do we actually want that RandomID? Seems unnecessarily long?
```
2020-03-18T16:29:41.963Z [INF] c:SGCollect-25ddbd1bf5d4ea655e3b0b4f1e6127c5 sgcollect_info: ntp time (ntpdate -q pool.ntp.org || nc time.nist.gov 13 || netcat time.nist.gov 13) - OK
```
Guess it doesn't really matter all that much though.